src/uu/uniq/src/uniq.rs

```diff
@@ -247,6 +250,40 @@ fn opt_parsed<T: FromStr>(opt_name: &str, matches: &ArgMatches) -> UResult<Optio
     })
 }
 
+// gets number of fields to be skipped from the shorthand option `-N`
+//
+//```
+//uniq -12345
+//```
+// the first digit isn't interpreted by clap as part of the value
+// so `get_one()` would return `2345`, then to get the actual value
+// we loop over every possible first digit, append the value to it
+// and parse the resulting string as usize, an error at this point
+// means that a character which isn't a digit was given
+fn obsolete_skip_field(matches: &ArgMatches) -> UResult<Option<usize>> {
+    for opt_text in OBSOLETE_SKIP_FIELDS_DIGITS {
+        let argument = matches.get_one::<String>(opt_text);
+        if matches.contains_id(opt_text) {
+            let mut full = opt_text.to_owned();
+            if let Some(ar) = argument {
+                full.push_str(ar);
+            }
+            let value = full.parse::<usize>();
+
+            if let Err(_reason) = value {
+                return Err(USimpleError {
+                    code: 42,
+                    message: format!("Invalid argument for skip-fields: {}", full),
+                }
+                .into());
+            } else if let Ok(val) = value {
+                return Ok(Some(val));
+            }
```

Comment:
Nit: A `match` seems a bit more appropriate here:
```suggestion
            return match value {
                Ok(val) => Ok(Some(val)),
                Err(_) => {
                    Err(USimpleError {
                        code: 42,
                        message: format!("Invalid argument for skip-fields: {}", full),
                    })
                    .into());
                }
            };
```

Or since you're discarding the error anyway:
```suggestion
            if let Ok(val) = value {
                return Ok(Some(val));
            } else {
                return Err(USimpleError {
                    code: 42,
                    message: format!("Invalid argument for skip-fields: {}", full),
                }
                .into());
            }
```